examples/java/src/test/java/org/apache/beam/examples/complete/TrafficRoutesIT.java

```diff
@@ -63,14 +69,22 @@ public void testE2ETrafficRoutes() throws Exception {
     this.options.setBigQueryDataset(this.outputDatasetId);
     this.options.setBigQueryTable(this.outputTable);
     TrafficRoutes.runTrafficRoutes(options);
-
-    QueryResponse response =
-        this.bqClient.queryWithRetries(
-            String.format(
-                "SELECT count(*) as total FROM [%s:%s.%s]",
-                this.projectId, this.outputDatasetId, this.outputTable),
-            this.projectId);
-    String res = response.getRows().get(0).getF().get(0).getV().toString();
+    FluentBackoff backoffFactory =
+        FluentBackoff.DEFAULT.withMaxRetries(4).withInitialBackoff(Duration.standardSeconds(1L));
+    Sleeper sleeper = Sleeper.DEFAULT;
+    BackOff backoff = BackOffAdapter.toGcpBackOff(backoffFactory.backoff());
+    String res = "empty_result";
+    do {
+      QueryResponse response =
+          this.bqClient.queryWithRetries(
+              String.format(
+                  "SELECT count(*) as total FROM [%s:%s.%s]",
+                  this.projectId, this.outputDatasetId, this.outputTable),
+              this.projectId);
+      if (response.getRows() != null && response.getRows().get(0).getF() != null) {
+        res = response.getRows().get(0).getF().get(0).getV().toString();
```

Comment:
We should terminate the `while` loop after we get the response

Reply:
Sorry for missing  a break.